API.md

```diff
@@ -17,20 +17,20 @@ Make a request to a Web2 Ethereum node using the caller's URL to an openly avail
 
 ```candid
 type Source = variant {
-  Url : text;
-  Service : record { hostname : text; chain_id : opt nat64 };
   Chain : nat64;
+  Service : record { hostname : text; chain_id : opt nat64 };
   Provider : nat64;
+  Custom : { url : text; headers : opt vec (text, text) };
 };
 
-request: (source: Source, json_rpc_payload: text, max_response_bytes: nat64) -> (Result<text, EthRpcError>);
+request: (source: Source, json_rpc_payload: text, max_response_bytes: nat64) -> (Result<text, RpcError>);
 ```
 
 * `source`: Any of the following enum variants:
   * `Chain: nat64` The relevant EVM network identifier ([reference list](https://chainlist.org/?testnets=true)).
   * `Service: { hostname: text, chain_id: opt nat64 }` A specific RPC service with the given hostname (e.g. `cloudflare-eth.com`) and optional chain id (`0x1` for Ethereum mainnet).
   * `Provider: nat64` The ID of the provider to be used for this call. Call `get_providers` to view a full list of providers.
-  * `Url: text` The URL of the service, including any API key if required for access-protected services. It's only recommended to use this if no registered RPC provider exists for a given EVM network.
+  * `Custom : { url : text; headers : opt vec (text, text) }` The URL of the service, including any API key if required for access-protected services. It's only recommended to use this if no registered RPC provider exists for a given EVM network.
```

Comment:
Where is the use of headers defined? And why is a header of type `(text, text)`?